app/src/main/java/io/zenandroid/onlinego/utils/WhatsNewUtils.kt

```diff
@@ -1,14 +1,35 @@
 package io.zenandroid.onlinego.utils
+import java.security.MessageDigest
 
 object WhatsNewUtils {
 
     val shouldDisplayDialog: Boolean
-            get() = PersistenceManager.lastWhatsNewText != currentText
-//            get() = PersistenceManager.lastWhatsNewText != null && PersistenceManager.lastWhatsNewText != currentText
+        get() = PersistenceManager.lastWhatsNewText != hashString(currentText)
+
     val whatsNewText = currentText
 
     fun textShown() {
-        PersistenceManager.lastWhatsNewText = currentText
+        PersistenceManager.lastWhatsNewText = hashString(currentText)
+    }
+
+    private fun hashString(text: String): String {
+        return toHex(MessageDigest.getInstance("MD5").digest(text.toByteArray(Charsets.UTF_8)))
```

Comment:
what I think you're trying to achieve with `toHex()` is basically `.fold("", { str, it -> str + "%02x".format(it) })` isn't it? If so, please replace it with that.

Reply:
nice one!